src/plugins/tangorin.rs

```diff
@@ -0,0 +1,214 @@
+use std::io;
+use irc::client::prelude::*;
+use regex::Regex;
+use plugin::Plugin;
+
+extern crate kuchiki;
+use kuchiki::traits::*;
+
+register_plugin!(Tangorin);
+
+lazy_static! {
+    static ref RE: Regex = Regex::new(r"!tangorin (\S+)").unwrap();
+}
+
+impl Tangorin {
+    fn grep_kanji(&self, msg: &str) -> Option<String> {
+        match RE.captures(msg) {
+            Some(captures) => {
+                Some(format!("{}", captures.at(1).unwrap()))
+            },
+            None => None
+        }
+    }
+
+    fn retrieve_from_selector(&self, doc: &kuchiki::NodeRef, selector: &str) -> Option<String> {
+        if let Some(match_) = doc.select(selector).unwrap().next() {
+            let node    = match_.as_node().first_child().unwrap();
+            let borrowed_text   = node.as_text().unwrap().borrow();
+            let mut retrieved_text = borrowed_text.clone();
```

Comment:
I think this doesn't need to be mutable as you should be able to do everything in a single line, right?